Packs/Citrix/Integrations/CitrixDaasEventCollector/CitrixDaasEventCollector.py

```diff
@@ -0,0 +1,370 @@
+    """Returns 0 if the timestamp is less than one hour old; otherwise,
+    returns the day difference (1 = today, 2 = yesterday, etc.)."""
+    # Parse the ISO-8601 timestamp with Zulu time (UTC)
+    dt = datetime.strptime(timestamp_str, API_RES_DATE_FORMAT)
+    dt = dt.replace(tzinfo=timezone.utc)
+
+    # Current time in UTC
+    now = datetime.now(timezone.utc)
+
+    delta = now - dt
+    # Difference in minutes
+    minutes = delta.total_seconds() / 60
+    # Difference in days
+    days = delta.days
+
+    if minutes < 60:
+        return 0
+
+    if days <= 0:
+        return 1
+
+    return delta.days + 1
+
+
+def deduplicate_events(events: list[dict[str, Any]], last_fetched_ids: list[str]) -> list[dict[str, Any]]:
+    """Remove already-processed events based on previously fetched IDs."""
+
+    if not last_fetched_ids:
+        demisto.debug("[Dedup] No deduplication needed (first run - no previous IDs)")
+        return events
+
+    demisto.debug(f"[Dedup] Checking {len(events)} events against {len(last_fetched_ids)} previously fetched IDs")
+
+    # Convert to set for O(1) lookup
+    fetched_ids_set = set(last_fetched_ids)
+
+    # Filter out events that were already fetched
+    new_events = [event for event in events if event.get("id") not in fetched_ids_set]
+
+    skipped_count = len(events) - len(new_events)
+    if skipped_count > 0:
+        demisto.debug(f"[Dedup] Skipped {skipped_count} duplicates. {len(new_events)} new events remain.")
+    else:
+        demisto.debug("[Dedup] No duplicates found.")
+
+    return new_events
+
+
+def fetch_events_command(client: Client, max_fetch: int, last_run: dict):
+    last_run_date = last_run.get("LastRun")
+    last_fetched_ids = last_run.get("LastFechedIds", [])
+    last_operation_id = last_fetched_ids[-1] if last_fetched_ids else None
+
+    if not last_run_date:
+        operations, _ = client.get_operations_with_pagination(
+            limit=max_fetch, last_operation_id=last_operation_id, search_date_option="LastMinute", last_run_date=last_run_date
+        )
+    else:
+        days = days_since(last_run_date)
+
+        if days == 0:
+            operations, _ = client.get_operations_with_pagination(
+                limit=max_fetch, last_operation_id=last_operation_id, search_date_option="LastHour", last_run_date=last_run_date
+            )
+        else:
+            operations, _ = client.get_operations_with_pagination(
+                limit=max_fetch, last_operation_id=last_operation_id, days=days, last_run_date=last_run_date
+            )
+
+    if operations:
+        # Deduplicate
+        operations = deduplicate_events(operations, last_fetched_ids)
+        new_last_run = operations[-1]["_time"]
+
+        ids_at_last_timestamp = [
+            operation.get("Id") for operation in operations if operation.get("_time") == new_last_run and operation.get("Id")
+        ]
+
+        last_run = {"LastRun": new_last_run, "LastFechedIds": ids_at_last_timestamp}
```

Comment:
If there are more events with the same timestamp than max_fetch, and they're all duplicates, the integration could get stuck in an infinite loop